src/metrics/codesmells_metrics.nit

```diff
@@ -0,0 +1,268 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+module codesmells_metrics
+
+# We usualy need specific phases
+# NOTE: `frontend` is sufficent in most case (it is often too much)
+import frontend
+
+import metrics_base
+import mclasses_metrics
+import semantize
+
+import method_analyse_metrics
+
+class BadConceptions
+	#Code smell list
+	var badConceptionElement = new Array[BadConception]
+
+	# Print all element conception
+	fun printAll do
+		for cd in badConceptionElement do
+			cd.printResult
+		end
+	end
+
+	#Collection
+	fun collect(classe : AClassdef)do 
+		for cd in badConceptionElement do
+			cd.collect(classe)
+		end 
+	end
+end
+
+class Antipatterns
+	super BadConceptions
+	# create all Antipatterns
+	init do
+		#badConceptionElement.add(new GOC)
+	end
+end
+
+class CodeSmells
+	super BadConceptions
+	# create all codesmells	
+	init do
+		badConceptionElement.add(new LARGC)
+		badConceptionElement.add(new LONGPL)
+		badConceptionElement.add(new FEM)
+		badConceptionElement.add(new LONGMETH)
+	end
+end
+
+
+
+class BadConception
+	#Bool Result
+	var result = false
+
+	#Name
+	fun name: String do 
+		return ""
+	end
+
+	#Description
+	fun desc: String do 
+		return ""
+	end
+
+	#Collection method
+	fun collect(classe : AClassdef)do end
+
+	#Show results in console
+	fun printResult do
+		print "{desc} :  {result}"
+	end
+end
+
+
+class Antipattern
+	super BadConception
+end
+
+
+class CodeSmell
+	super BadConception
+end
+
+
+class LARGC
+	super CodeSmell
+
+	redef fun name do 
+		return "LARGC"
+	end
+	redef fun desc do 
+		return "Large class"
+	end
+
+	redef fun collect(aclasse) do
+
+		#get class definition 
+		var classe = aclasse.mclassdef
+		var numberAttribut = classe.collect_intro_mattributes.length
+		#get the number of methods and subtract the get and set of attibutes (numberAtribut*2)
+		var numberMethode = classe.collect_intro_mattributes.length - (numberAttribut*2)
+		if numberMethode > 20 or numberAttribut > 20 then result = true
+	end
+end
+
+class LONGPL
+	super CodeSmell
+
+	var badmethods = new Array[MMethodDef] 
+
+	redef fun name do 
+		return "LONGPL"
+	end
+	redef fun desc do 
+		return "Long parameter list"
+	end
+
+	redef fun collect(aclasse) do
```

Comment:
In this case, why not name it `aclass` instead of `aclasse`?
Sorry to be be picky with that, but I like code to be consistent in the way things are named